trulens_eval/trulens_eval/feedback.py

```diff
@@ -1,5 +1,353 @@
+specified by the `on...` methods of the `Feedback` objects. The general form is:
+
+```python
+feedback: Feedback = feedback.on(argname1=selector1, argname2=selector2, ...)
+```
+
+That is, `Feedback.on(...)` returns a new `Feedback` object with additional
+argument mappings, the source of `argname1` is `selector1` and so on for further
+argument names. The types of `selector1` is `JSONPath` which we elaborate on in
+the further sections.
+
+If argument names are ommitted, they are taken from the feedback function
+implementation signature in order. That is, 
+
+```python
+...on(argname1=selector1, argname2=selector2)
+```
+
+and
+
+```python
+...on(selector1, selector2)
+```
+
+are equivalent assuming the feedback implementation has two arguments,
+`argname1` and `argname2`, in that order.
+
+Several utility methods starting with `.on` provide shorthands:
+
+- `on_input(arg) == on_prompt(arg: Optional[str])` -- both specify that the next
+    unspecified argument or `arg` should be the main app input.
+
+- `on_output(arg) == on_response(arg: Optional[str])` -- specify that the next
+    argument or `arg` should be the main app output.
+
+- `on_input_output() == on_input().on_output()` -- specifies that the first
+    two arguments of implementation should be the main app input and main app
+    output, respectively.
+
+- `on_default()` -- depending on signature of implementation uses either
+    `on_output()` if it has a single argument, or `on_input_output` if it has
+    two arguments.
+
+Some wrappers include additional shorthands:
+
+### llama_index-specific selectors
+
+- `TruLlama.select_source_nodes()` -- outputs the selector of the source
+    documents part of the engine output.
+
+### Running Feedback
+
+Feedback implementations are simple callables that can be run on any arguments
+matching their signatures. However, once wrapped with `Feedback`, they are meant
+to be run on outputs of app evaluation (the "Records"). Specifically,
+`Feedback.run` has this definition:
+
+```python
+def run(self, 
+    app: Union[AppDefinition, JSON], 
+    record: Record
+) -> FeedbackResult:
+```
+
+That is, the context of a Feedback evaluation is an app (either as
+`AppDefinition` or a JSON-like object) and a `Record` of the execution of the
+aforementioned app. Both objects are indexable using "Selectors". By indexable
+here we mean that their internal components can be specified by a Selector and
+subsequently that internal component can be extracted using that selector.
+Selectors for Feedback start by specifying whether they are indexing into an App
+or a Record via the `__app__` and `__record__` special
+attributes (see **Selectors** section below).
+
+#### App/Record Organization
+
+Apps are serialized into JSON-like structures which are indexed via selectors.
+The exact makeup of this structure is app-dependent though always start with
+`app`, that is, the trulens wrappers (subtypes of `App`) contain the wrapped app
+in the attribute `app`:
+
```

Comment:
add: 

to see your app and selection structure
```
tru.get_app(app_name="..")
```

Reply:
Can we just give a simple example right here?

If your app looks something like 
```
fill in here
```

Then your selector would look like this

```
Feedback(feedback_fn).on(Select.on(this.path.relevent.to.example))
```

Reply:
Problem with this is that most examples do not make the path clear as they are hidden in the langchain / llamaindex details.

For now I did add more text as to how to see the app structure that can be indexed.